state/application.go

```diff
@@ -1726,13 +1724,9 @@ func (a *Application) addUnitOpsWithCons(args applicationAddUnitOpsArgs) (string
 	// history entries. This is risky, and may lead to extra entries, but that's
 	// an intrinsic problem with mixing txn and non-txn ops -- we can't sync
 	// them cleanly.
-	if unitStatusDoc != nil {
```

Comment:
Again here, all of these conditions are always true.

We could follow the logic in these methods better is we didn't let them get so big.